src/extra.ml

```diff
@@ -128,6 +188,22 @@ module Array =
         elements with [eq]. *)
     let equal : 'a eq -> 'a array eq = fun eq a1 a2 ->
       Array.length a1 = Array.length a2 && for_all2 eq a1 a2
+
+    (** [argmax ?init c a] finds the index of the maximum according to
+        comparison function [c] in array [a] with initial value [?init].  If
+        [?init] is not provided, the first element of [a] is used. *)
+    let argmax : ?init:'a -> 'a eq -> 'a array -> int = fun ?init cmp arr ->
```

Comment:
Rename cmp into eq to avoir confusion with comparison functions with value in int in OCaml.